monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/data/manager/InMemoryDataManager.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ *
+ */
+
+package org.wso2.ei.dashboard.core.data.manager;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.wso2.ei.dashboard.core.exception.DashboardServerException;
+import org.wso2.ei.dashboard.core.rest.delegates.heartbeat.HeartbeatObject;
+import org.wso2.ei.dashboard.core.rest.model.GroupList;
+import org.wso2.ei.dashboard.core.rest.model.NodeList;
+import org.wso2.ei.dashboard.core.rest.model.NodeListInner;
+import java.util.HashMap;
+
+/**
+ * Performs in-memory map operations.
+ */
+public final class InMemoryDataManager implements DataManager {
+
+    private static final Logger logger = LogManager.getLogger(InMemoryDataManager.class);
+    public static HashMap<String, HashMap> heartBeatStore;
+    public static HashMap<String, HashMap> serviceInfoStore;
+
+    public InMemoryDataManager() {
+        heartBeatStore = new HashMap<>();
+        serviceInfoStore = new HashMap<>();
+        logger.debug("heartBeatStore and serviceInfoStore created ");
+    }
+
+    @Override
+    public boolean insertHeartbeat(HeartbeatObject heartbeat, String accessToken) {
+        try {
+            HashMap map = new HashMap<>();
+            map.put("groupId", heartbeat.getGroupId());
```

Comment:
Can the HeartbeatObject become null? If so better to handle it.

Reply:
According to 

https://github.com/wso2/micro-integrator/blob/8d6f83be5bbbf74675752fce1f949dfa2b133cad/components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/dashboard/HeartBeatComponent.java#L154, 

https://github.com/wso2/micro-integrator/blob/8d6f83be5bbbf74675752fce1f949dfa2b133cad/components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/dashboard/HeartBeatComponent.java#L165, 

https://github.com/wso2/micro-integrator/blob/8d6f83be5bbbf74675752fce1f949dfa2b133cad/components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/dashboard/HeartBeatComponent.java#L178

https://github.com/wso2/micro-integrator/blob/8d6f83be5bbbf74675752fce1f949dfa2b133cad/components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/dashboard/HeartBeatComponent.java#L132

these values cannot be null. But this will be an issue once we extend this to other services such as CLI. 

Therefore let's add a null check. Shall we add a null check in https://github.com/wso2/product-mi-tooling/blob/5af2287b6de4701bcd845b7a1457d2fc357b3ade/monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/core/rest/delegates/heartbeat/HeartBeatDelegate.java#L53-L58. Prior to heartbeat object creation.